peer/peer.go

```diff
@@ -82,11 +82,6 @@ var (
 	// sentNonces houses the unique nonces that are generated when pushing
 	// version messages that are used to detect self connections.
 	sentNonces = newMruNonceMap(50)
-
-	// allowSelfConns is only used to allow the tests to bypass the self
-	// connection detecting and disconnect logic since they intentionally
-	// do so for testing purposes.
-	allowSelfConns bool
```

Comment:
With `TstAllowSelfConns` defined in `export_test.go` in the `peer` package, it can't be called from tests in the `netsync` package. If I run `go test ./netsync/...`, I get `undefined: peer.TstAllowSelfConns`.

If you really don't like having it in the peer config, I can move `TstAllowSelfConns()` into the `peer` package proper (from export_test). However, this pattern still feels kind of wrong to me. First, the global doesn't get unset currently, so if it is running multiple tests the change of this global setting could leak into other tests I believe. What might be better is on a per-test basis to call:

```go
func TestSomething(t *testing.T) {
  peer.TstAllowSelfConns()
  defer peer.TstDisallowSelfConns()

  ...
}
```